src/retsnoop.c

```diff
@@ -196,9 +204,55 @@ static int append_str_file(char ***strs, int *cnt, const char *file)
 	return err;
 }
 
-static int append_compile_unit(char ***strs, int *cnt, const char *compile_unit)
+static int append_compile_unit(struct ctx *ctx, char ***strs, int *cnt, const char *compile_unit)
 {
 	int err = 0;
+
+	int max_resp = 128;
+	struct a2l_cu_resp *cu_resps =
+		(struct a2l_cu_resp*)malloc(sizeof(struct a2l_cu_resp) * max_resp);
+	if (cu_resps == NULL) {
+		return -ENOMEM;
+	}
+
+	int resp_cnt = addr2line__query_symbols(ctx->a2l, compile_unit, cu_resps, max_resp);
```

Comment:
this `max_resp` restriction is not great. Let's make query_symbols allocate enough memory to contain all the results and let retsnoop part free this memory at the end?

128 might be too small for some files, but there is also no good reason to artificially limit this